data_structures/src/Stack/Stack.cs

```diff
@@ -6,44 +6,71 @@ namespace ADT
 {
     public class Stack<T> : IStack<T>
     {
+        private T[] _stack { get; set; }
+        private int _lastIndex { get; set; }
+
         public Stack(int size)
         {
-
+            _stack = new T[size]; 
         }
 
         public IEnumerator<T> GetEnumerator()
         {
-            throw new NotImplementedException();
+             return (IEnumerator<T>)_stack.GetEnumerator(); 
         }
 
         public bool IsEmpty()
         {
-            throw new NotImplementedException();
+            return this.Size() > 0 ? false : true;
```

Comment:
No need for the `this` keyword, it is noisy and redundant in a language like c#. You also don't need this ternary expression; `return this.Size() > 0` is sufficient.